### ISSUE TITLE
Should raise `undefined-variable` for self-referencing walrus operator expressions

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5653]
### Bug description

```python
"""Should raise undefined-variable"""
if var := var:
    print(var)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

### Expected behavior

line 2: undefined-variable

### Pylint version

```shell
pylint 2.13.0-dev0
astroid 2.9.2
Python 3.10.1 (v3.10.1:2cd268a3a9, Dec  6 2021, 14:28:59) [Clang 13.0.0 (clang-1300.0.29.3)]
```
